Setup steps for Cilium CNI

ackerman80 opened the pull request:
Link to guide for setting up Cilium CNI with kubeadm.

ackerman80 commented:
Signed the CLA as a employee contributor part of the Covalent CNCF team

On Thu, Aug 9, 2018 at 2:02 PM, k8s-ci-robot <notifications@github.com>
wrote:

> Thanks for your pull request. Before we can look at your pull request,
> you'll need to sign a Contributor License Agreement (CLA).
>
> 📝 *Please follow instructions at
> https://git.k8s.io/community/CLA.md#the-contributor-license-agreement
> <https://git.k8s.io/community/CLA.md#the-contributor-license-agreement> to
> sign the CLA.*
>
> It may take a couple minutes for the CLA signature to be fully registered;
> after that, please reply here with a new comment and we'll verify. Thanks.
> ------------------------------
>
>    - If you've already signed a CLA, it's possible we don't have your
>    GitHub username or you're using a different email address. Check your
>    existing CLA data and verify that your email is set on your git commits
>    <https://help.github.com/articles/setting-your-email-in-git/>.
>    - If you signed the CLA as a corporation, please sign in with your
>    organization's credentials at https://identity.
>    linuxfoundation.org/projects/cncf
>    <https://identity.linuxfoundation.org/projects/cncf> to be authorized.
>    - If you have done the above and are still having issues with the CLA
>    being reported as unsigned, please email the CNCF helpdesk:
>    helpdesk@rt.linuxfoundation.org <helpdesk@rt.linuxfoundation.org>
>
> Instructions for interacting with me using PR comments are available here
> <https://git.k8s.io/community/contributors/guide/pull-requests.md>. If
> you have questions or suggestions related to my behavior, please file an
> issue against the kubernetes/test-infra
> <https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:>
> repository. I understand the commands that are listed here
> <https://go.k8s.io/bot-commands>.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/website/pull/9806#issuecomment-411896064>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AWSTV_9go1imdrJ5J8LfoOXiu8nCpaFmks5uPKNwgaJpZM4V2qrm>
> .
>

neolit123 commented:
@ackerman80 

thank you for the PR.
please squash the commits into a single one.

i'm getting the following message when i try to open the linked website:
![error](https://user-images.githubusercontent.com/331852/43930423-f9db457c-9c41-11e8-892f-704ec230e9a0.png)

i doubt that this will be accepted as is!

/assign @Bradamant3 @tengqm 
/cc @xiangpengzhao @dixudx 
/cc @kubernetes/sig-cluster-lifecycle-pr-reviews

neolit123 commented on content/en/docs/setup/independent/create-cluster-kubeadm.md:
```diff
@@ -277,8 +277,14 @@ For Canal to work correctly, `--pod-network-cidr=10.244.0.0/16` has to be passed
+
+{{% tabe name="Cilium" %}}
+
+Cilium provides networking, L3/L4 network security and L7 (API-aware) security. Cilium is deployed as a CNI plugin and supports the Kubernetes Network Policy specifications. Please consult the [Cilium Kubernetes Install Guide](https://docs.cilium.io/en/stable/kubernetes/install/) for complete steps including RBAC settings, etcd access config and Kubernetes version specific daemonset configuration ...
```
`daemonset` -> `DaemonSet`

xiangpengzhao commented:
For the link issue, I guess "docs.cilium.io" is using "readthedocs.io" framework and its certificate then the browser can't verify the cert for "docs.cilium.io".
/cc @zacharysarah @steveperry-53 @Bradamant3

ackerman80 commented:
I've changed the link to https://cilium.readthedocs.io/en/stable/kubernetes/install/  as the certificate is for *.readthedocs.io should work. Also squashed the commits and made the other suggested edits. cc @xiangpengzhao @neolit123

neolit123 commented on content/en/docs/setup/independent/create-cluster-kubeadm.md:
```diff
@@ -277,6 +277,11 @@ For Canal to work correctly, `--pod-network-cidr=10.244.0.0/16` has to be passed
+
+{{% tabe name="Cilium" %}}
+
+Cilium provides networking, L3/L4 network security and L7 (API-aware) security. Cilium is deployed as a CNI plugin and supports the Kubernetes Network Policy specifications. Please consult the [Cilium Kubernetes Install Guide](https://cilium.readthedocs.io/en/stable/kubernetes/install/) for complete steps including RBAC settings, etcd access configuration and Kubernetes version specific daemonset ...
```
`daemonset` -> `DaemonSet`

neolit123 replied:
other than that LGTM but would leave the approval to others.

ackerman80 commented:
Signed the CLA as a employee contributor part of the Covalent CNCF team

On Thu, Aug 9, 2018 at 10:36 PM, k8s-ci-robot <notifications@github.com>
wrote:

> Thanks for your pull request. Before we can look at your pull request,
> you'll need to sign a Contributor License Agreement (CLA).
>
> 📝 *Please follow instructions at
> https://git.k8s.io/community/CLA.md#the-contributor-license-agreement
> <https://git.k8s.io/community/CLA.md#the-contributor-license-agreement> to
> sign the CLA.*
>
> It may take a couple minutes for the CLA signature to be fully registered;
> after that, please reply here with a new comment and we'll verify. Thanks.
> ------------------------------
>
>    - If you've already signed a CLA, it's possible we don't have your
>    GitHub username or you're using a different email address. Check your
>    existing CLA data and verify that your email is set on your git commits
>    <https://help.github.com/articles/setting-your-email-in-git/>.
>    - If you signed the CLA as a corporation, please sign in with your
>    organization's credentials at https://identity.
>    linuxfoundation.org/projects/cncf
>    <https://identity.linuxfoundation.org/projects/cncf> to be authorized.
>    - If you have done the above and are still having issues with the CLA
>    being reported as unsigned, please email the CNCF helpdesk:
>    helpdesk@rt.linuxfoundation.org <helpdesk@rt.linuxfoundation.org>
>
> Instructions for interacting with me using PR comments are available here
> <https://git.k8s.io/community/contributors/guide/pull-requests.md>. If
> you have questions or suggestions related to my behavior, please file an
> issue against the kubernetes/test-infra
> <https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:>
> repository. I understand the commands that are listed here
> <https://go.k8s.io/bot-commands>.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/website/pull/9806#issuecomment-411980297>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AWSTV-lvfQU0aj3Im-ffL771bHxClXR1ks5uPRvBgaJpZM4V2qrm>
> .
>

neolit123 commented:
/check-cla

ackerman80 commented:
/check-cla

neolit123 commented:
@ackerman80 
if you have issues with the CLA contact the CNCF:
info@cncf.io
(or via a help desk if they have one.)

ackerman80 commented:
Yup sent a note to  helpdesk@rt.linuxfoundation.org


On Fri, Aug 10, 2018 at 10:07 AM, Lubomir I. Ivanov <
notifications@github.com> wrote:

> @ackerman80 <https://github.com/ackerman80>
> if you have issues with the CLA contact the CNCF:
> info@cncf.io
> (or via a help desk if they have one.)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/website/pull/9806#issuecomment-412145629>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AWSTV-GqYWImwl8IPn8VBIX_PYYDNr6Cks5uPb3ogaJpZM4V2qrm>
> .
>

arvindsoni80 commented:
/check-cla

neolit123 commented:
thank for addressing the CLA issue.
this is LGTM, waiting on feedback from the SIG and we can possibly merge it.

zacharysarah commented:
/assign @codyclark

zacharysarah commented:
/assign @cody-clark

zacharysarah commented:
@cody-clark Looks like we need to fast-track your org membership!

ackerman80 commented:
@xiangpengzhao @neolit123 @zacharysarah is there something blocking to get this merged?

xiangpengzhao commented on content/en/docs/setup/independent/create-cluster-kubeadm.md:
````diff
@@ -277,6 +277,11 @@ For Canal to work correctly, `--pod-network-cidr=10.244.0.0/16` has to be passed
 ```
+{{% /tab %}}
+
+{{% tabe name="Cilium" %}}
````
Netlify failed with this error:
```
11:28:05 PM: ERROR 2018/08/10 15:28:05 Unable to locate template for shortcode "tabe" in page "docs/setup/independent/create-cluster-kubeadm.md"
```

I think it's due to a typo here: `s/tabe/tab`

ackerman80 commented:
@xiangpengzhao fixed the typo. thanks for catching that

xiangpengzhao commented:
/lgtm
Let other guys give a final approval if no more change needed :)

neolit123 commented:
a more detailed PR was submitted for Cilium here.
@ackerman80 please combine your efforts with that PR submitter and review his changes.
https://github.com/kubernetes/website/pull/9889
/close
